### PR TITLE
Feat/add exercises

### PR DESCRIPTION
**Pull request by doctype-Melvin**
PR for [US4](https://github.com/doctype-Melvin/capstone/issues/8)

- Users can create a plan
- Users can add and update exercises
- there's a collapsible component (clunky)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+    "ecmaVersion": 12,
+    "sourceType": "module"
   },
-  "plugins": ["jest"]
+  "plugins": ["react"],
+  "rules": {
+    "react/react-in-jsx-scope": "on",
+    "react/prop-types": 0
+  }
 }
```
Comment (djfarly) [inline .eslintrc.json]: why the change here?

The default eslint config should be fine: https://github.com/neuefische/web-exercises/blob/main/templates/next/.eslintrc.json

File: components/Collapsible/Collapsible.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+`;
+
+export default function Collapsible({
+  id,
+  sets,
+  reps,
+  weight,
+  dayId,
+  handleUpdateExercise,
+}) {
```
Comment (djfarly) [inline components/Collapsible/Collapsible.js]: This component does much more than just handling collapsible content. Consider splitting this into multiple parts (Collapsible, ExerciseDetail, etc.)

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Melvin,

thank you for the PR. 🥳 

I think you are doing too many data/state related things that contradict each other in this PR. As it is, it is very hard to follow the logic of your app.

1. A database + API + `useSWR`
2. localStorage (both via the custom hook and via the standard api)
3. Context

If you want to use a database you'll have to go all in on that approach. That means: Getting rid of localStorage and Context. Both things are complicating your App quite a lot.

You can use your `useSWR` hook at any level of your app to retrieve plan data without the need for Context or localStorage. To make this easier you could wrap it inside a custom hook alà

```js
function usePlan(id) {
  return useSWR(`/api/plans/${id}`, fetcher);
}
```

You are using the local copy of your state to update your data. This does not make much sense because the source of truth for your data should be the database. Thus you'll need to implement create, edit and delete features for your API routes and implement SWRMutations to mutate the result of the fetch.

tl;dr: Please do not use localStorage and Context for the same data that also exists in your database.

---

Please format your code using prettier. Setup VSCode to use Prettier as your default formatter and enable "Format on Save" in the settings.

You can run the following command in your project directory to format all of your code:
```
npx -y prettier@latest --write .
```

---

I really hope my large blob of text does not feel discouraging. Keep on hacking!



Cheers!

File: .eslintrc.json (hunk 1 above)
Comment (doctype-Melvin) [reply .eslintrc.json]: The Formatter never worked to begin with. I tried a couple of things but to no avail. It formatted some parts of my code on save once. 
I've changed my eslintrc back to the default config now.

File: components/Collapsible/Collapsible.js (hunk 2 above)
Comment (doctype-Melvin) [reply components/Collapsible/Collapsible.js]: Thanks. I'll look into it.

**Comment by doctype-Melvin** [comment]
First things first - thanks for taking the time to review my PR!
To be honest, I didn't feel comfortable pushing this PR into review. Mostly because of what you're pointing out. I first thought it would be a great idea to create the plan in my database to have the mongodb id available.

But then I thought about users creating and changing their exercises - maybe even frequently - ~~would take a tad bit longer if using the database connection compared to browser storage.~~ Hence, I chose to manage all the updating using local storage. My next user story would've been about submitting the completed plan to the database.

Considering that no one ever will use this app, I could've just kept on rolling with the database route.
Just wanted to sketch out my reasoning for this PR.

Again, happy to read your feedback. Not discouraging at all.

Edit: I think I now have a better understanding of the useSWR hook. Thanks.

**Comment by frank12421** [comment]
Hi Melvin, this looks very good. 
Your code is built very understandable. All the suggestions have been implemented. I look forward to seeing and learning more from you.

**Comment by doctype-Melvin** [comment]
Thank you Frank, and thanks for taking the time to review my code. 
Happy to read that you could follow my code, since writing clean code is my biggest challenge really.
